src/api/async_resource.cc

```diff
@@ -16,7 +16,7 @@ AsyncResource::AsyncResource(Isolate* isolate,
                              Local<Object> resource,
                              const char* name,
                              async_id trigger_async_id)
-    : env_(Environment::GetCurrent(isolate)),
+    : env_(Environment::GetCurrent(isolate->GetCurrentContext())),
```

Comment:
For example, this is not doing what the PR description claims. The code here still needs the Environment, therefore it still needs to look it up from the context, therefore it still needs a HandleScope. It's not up to the caller to choose to create it or not - if it doesn't, V8 can just crash.